src/java.base/share/classes/java/lang/FdLibm.java

```diff
@@ -504,6 +504,451 @@ static double compute(double y, double x) {
+
+        static double compute(double x) {
+            double z = 0.0;
+            int sign = 0x8000_0000;
+            /*unsigned*/ int r, t1, s1, ix1, q1;
+            int ix0, s0, q, m, t, i;
+
+            ix0 = __HI(x);  // high word of x
+            ix1 = __LO(x);  // low word of x
+
+            // take care of Inf and NaN
+            if((ix0 & 0x7ff0_0000) == 0x7ff0_0000) {
+                return x*x + x; // sqrt(NaN)=NaN, sqrt(+inf)=+inf, sqrt(-inf)=sNaN
+            }
+            // take care of zero
+            if (ix0 <= 0) {
+                if (((ix0 & (~sign)) | ix1) == 0)
+                    return x; // sqrt(+-0) = +-0
+                else if (ix0 < 0)
+                    return (x-x)/(x-x); // sqrt(-ve) = sNaN
+            }
+            // normalize x
+            m = (ix0 >> 20);
+            if (m == 0) { // subnormal x
+                while (ix0 == 0) {
+                    m -= 21;
+                    ix0 |= (ix1 >>> 11); // unsigned shift
+                    ix1 <<= 21;
+                }
+                for(i = 0; (ix0 & 0x0010_0000) == 0; i++) {
+                    ix0 <<= 1;
+                }
+                m -= i-1;
+                ix0 |= (ix1 >>> (32 - i)); // unsigned shift
+                ix1 <<= i;
+            }
+            m -= 1023;      // unbias exponent */
+            ix0 = (ix0 & 0x000f_ffff) | 0x0010_0000;
+            if ((m & 1) != 0){        // odd m, double x to make it even
+                ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+                ix1 += ix1;
+            }
+            m >>= 1;        // m = [m/2]
+
+            // generate sqrt(x) bit by bit
+            ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+            ix1 += ix1;
+            q = q1 = s0 = s1 = 0;   // [q,q1] = sqrt(x)
+            r = 0x0020_0000;        // r = moving bit from right to left
+
+            while(r != 0) {
+                t = s0 + r;
+                if(t <= ix0) {
+                    s0   = t + r;
+                    ix0 -= t;
+                    q   += r;
+                }
+                ix0 += ix0 + ((ix1 & sign) >>> 31); // unsigned shift
+                ix1 += ix1;
+                r >>>= 1; // unsigned shift
+            }
+
+            r = sign;
+            while (r != 0) {
+                t1 = s1 + r;
+                t  = s0;
+                if((t < ix0) ||
+                   ((t == ix0) && (Integer.compareUnsigned(t1, ix1) <= 0 ))) { // t1 <= ix1
+                    s1  = t1 + r;
+                    if (((t1 & sign) == sign) && (s1 & sign) == 0) {
+                        s0 += 1;
+                    }
+                    ix0 -= t;
+                    if (Integer.compareUnsigned(ix1, t1) < 0) {  // ix1 < t1
+                        ix0 -= 1;
+                    }
+                    ix1 -= t1;
+                    q1  += r;
+                }
+                ix0 += ix0 + ((ix1 & sign) >>> 31);
```

Comment:
You may want to add the traditional `// unsigned shift` comment even here, just to be consistent.